Makefile

```diff
@@ -1,43 +1,39 @@
-_GOPATH 			:= $(PWD)/../../../..
-PROTOC := /usr/bin/protoc
-
-export GOPATH := $(_GOPATH)
-export PATH := $(_GOPATH)/bin:$(PATH)
+PROTOC := protoc
 
 .PHONY: all
 all: codegen
 
-$(_GOPATH)/src/github.com/gogo/protobuf/protoc-gen-gogoslick:
+$(GOPATH)/src/github.com/gogo/protobuf/protoc-gen-gogoslick:
```

Comment:
I recall reading something recently that said it's not necessary to set GOPATH in the environment anymore.  As people adopt this recommendation they may not have this environment variable set which would cause this Makefile to fail.  I think we should not require people to set GOPATH if we can avoid it.  It's possible to get the GOPATH using `go env GOPATH` even when the environment variable is not set.

And I think we can use `?=` to set it only if it's not already set.

So I think if we put this at the top of the Makefile we should be good.

```
GOPATH ?= $(shell go env GOPATH)
```

Reply:
Makefile was removed completely..
Replaced by Earthly..

Its 100% reproducible and just need 1 tool..
Will add readme entries how to use it..